platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -681,28 +720,57 @@ export default class HangingProtocolService extends PubSubService {
    * simpler than recomputing the entire protocol.
    */
   public getComputedOptions(
-    options: Record<string, unknown>,
+    options: Record<string, unknown> | Array<Record<string, unknown>>,
     displaySetUIDs: string[]
-  ) {
-    const computed = { ...options };
-    let displaySets;
-    for (const key in computed) {
-      const value = computed[key];
-      if (!value) continue;
-      if (value.custom) {
-        if (!displaySets) {
-          displaySets = this.displaySets.filter(
-            displaySet =>
-              displaySetUIDs.indexOf(displaySet.displaySetInstanceUID) !== -1
+  ): any {
+    // Base case: if options is an array, map over the array and recursively call getComputedOptions
+    if (Array.isArray(options)) {
+      return options.map(option =>
+        this.getComputedOptions(option, displaySetUIDs)
+      );
+    }
+
+    if (typeof options === 'object' && options !== null) {
+      // If options is an object with a custom attribute, compute a new options object
+      if ('custom' in options) {
```

Comment:
Prefer options.custom because what you actually care about is whether it is set to a value, and 0 doesn't count.  The in test can be useful at times, but isn't the right choice here.